nautobot_ssot/integrations/slurpit/signals.py

```diff
@@ -0,0 +1,106 @@
+    if cf_type == "type_date":
+        custom_field, _ = CustomField.objects.get_or_create(
+            key=key,
+            type=CustomFieldTypeChoices.TYPE_DATE,
+            label=label,
+        )
+    else:
+        custom_field, _ = CustomField.objects.get_or_create(
+            key=key,
+            type=CustomFieldTypeChoices.TYPE_TEXT,
+            label=label,
+        )
+    for model in models:
+        custom_field.content_types.add(ContentType.objects.get_for_model(model))
+    custom_field.save()
+
+
+def nautobot_database_ready_callback(sender, *, apps, **kwargs):  # pylint: disable=unused-argument
+    """Callback function triggered by the nautobot_database_ready signal when the Nautobot database is fully ready."""
+    # pylint: disable=invalid-name, too-many-locals
+    Device = apps.get_model("dcim", "Device")
+    DeviceType = apps.get_model("dcim", "DeviceType")
+    Manufacturer = apps.get_model("dcim", "Manufacturer")
+    Platform = apps.get_model("dcim", "Platform")
+    Role = apps.get_model("extras", "Role")
+    InventoryItem = apps.get_model("dcim", "InventoryItem")
+    Interface = apps.get_model("dcim", "Interface")
+    IPAddress = apps.get_model("ipam", "IPAddress")
+    Location = apps.get_model("dcim", "Location")
+    VLAN = apps.get_model("ipam", "VLAN")
+    VRF = apps.get_model("ipam", "VRF")
+    Prefix = apps.get_model("ipam", "Prefix")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    location_type = apps.get_model("dcim", "LocationType")
+    Tag = apps.get_model("extras", "Tag")
+
+    synced_tag, _ = Tag.objects.get_or_create(
+        name="SSoT Synced from Slurpit",
+        defaults={
+            "description": "Object synced at some point from Slurpit to Nautobot",
+            "color": ColorChoices.COLOR_LIGHT_GREEN,
+        },
+    )
+    synced_tag.content_types.add(ContentType.objects.get_for_model(Device))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(DeviceType))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(InventoryItem))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(Interface))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(IPAddress))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(Location))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(VLAN))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(VRF))
+    synced_tag.content_types.add(ContentType.objects.get_for_model(Prefix))
+
+    Tag.objects.get_or_create(
+        name="SSoT Safe Delete",
+        defaults={
+            "description": "Safe Delete Mode tag to flag an object, but not delete from Nautobot.",
+            "color": ColorChoices.COLOR_RED,
+        },
+    )
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    loc_type = location_type.objects.update_or_create(name="Site")[0]
```

Comment:
We've been moving away from creating a Region/Site LocationType for users and instead allowing the user to create whatever LocationType they want to associate for Device Locations and specifying that in the Job form. Nothing wrong with using Site but something to consider.